src/routes/notifications.js

```diff
@@ -0,0 +1,56 @@
+const express = require('express')
+const { body } = require('express-validator');
+
+const db = require('../db')
+const resolveAuthSettings = require('../utils/auth')
+const validate = require("../utils/validate");
+
+const createNotificationQuery = 'INSERT INTO notifications(type, payload, user_id, project_id) VALUES ($1, $2, $3, $4) returning *';
+const requiredMessage = 'This field is required and cannot be empty.';
+
+const router = express.Router()
+
+const auth = async function (req, res, next) {
+  try {
+    const settings = await resolveAuthSettings(req);
+
+    if (settings.role !== 'system_user') {
+      return res.status(403).json({ errors: [`User ${settings['request.user_id']} doesn't have permission to create notifications`] });
+    }
+
+    if (settings['request.project_id'] !== req.body['project_id']) {
+      return res.status(403).json({ errors: [
+        `User ${settings['request.user_id']} doesn't have permission to create notifications in project ${req.body['project_id']}`]
+      });
+    }
+
+    return next();
+  } catch (err) {
+    console.error(err)
+    return res.status(401).json({ errors: [err.message] });
+  }
+}
+
+router.use(auth)
+
+router.post('/',
+  validate([
+    body('type').notEmpty().withMessage(requiredMessage),
```

Comment:
Not sure, if it makes sense to add more strict validation, which would compare event type with some `supported_event_types`? I mean, will it break the system, if someone push unknown event type (ref templates, translations, etc.)